docs/replication-retry.md

```diff
@@ -0,0 +1,134 @@
+# Replication Retry
+
+## Description
+
+This feature offers a way for users to monitor and retry failed replication
+operations. It enables users to retrieve a list of failed operations and retry
+specific replication operations.
+
+## Requirements
+
+* The listing of failed operations.
+* The retry of a specific failed operation.
+
+## Design
+
+A RESTful API will expose methods for users to list and retry failed operations.
+See [Explanations](#explanations) for more detail.
+
+## Definition of API
+
+* GET `/_/crr/list/failed/all`
+
+    This GET request retrieves a listing of all failed operations. This
+    operation is useful if you're interested in whether any replication
+    operations have failed and want the entire listing.
+
+    Response:
+
+    ```sh
+    [{
+        bucket: <bucket>,
+        key: <key>,
+        versionId: <versionId>,
+        site: <site>
+    }]
+    ```
+
+* GET `/_/crr/list/failed/<bucket>/<key>/<versionId>`
+
+    This GET request retrieves a listing of all failed operations for a specific
+    object version. This operation is useful if you're interested in monitoring
+    the replication status of a specific object.
+
+    Response:
+
+    ```sh
+    [{
+        bucket: <bucket>,
+        key: <key>,
+        versionId: <versionId>,
+        site: <site>
+    }]
+    ```
+
+* POST `/_/crr/retry`
+
+    This POST request retries a failed operation.
+
+    Request Body:
+
+    ```sh
+    {
+        bucket: <bucket>,
+        key: <key>,
+        versionId: <versionId>,
+        site: <site>
+    }
+    ```
+
+    Response:
+
+    ```sh
+    {
+        site: <site>,
```

Comment:
Did you mean to also output the original params like `bucket`, `key` and `versionId` here?

Reply:
I figured this would be redundant, but with the change to support multiple retries in the POST request, it makes sense to include those params. I will do so, and update the response to be a list of retries as well since it could be multiple retries.